File: InPlan.Web/Modules/ServerTypes/Parametros/TecnicosRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface TecnicosRow {
    IdTecnico?: number;
    IdEmpresa?: number;
    IdArea?: number;
    NombreCompleto?: string;
    Celular?: string;
    Email?: string;
    Observaciones?: string;
    Baja?: boolean;
    IdEmpresaDescripcionCorta?: string;
    IdAreaDescripcion?: string;
}

export abstract class TecnicosRow {
    static readonly idProperty = 'IdTecnico';
    static readonly nameProperty = 'NombreCompleto';
    static readonly localTextPrefix = 'Parametros.Tecnicos';
    static readonly lookupKey = 'Parametros.Tecnicos';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TecnicosRow>('Parametros.Tecnicos') }
    static async getLookupAsync() { return getLookupAsync<TecnicosRow>('Parametros.Tecnicos') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<TecnicosRow>();
}

File: InPlan.Web/Modules/Parametros/Tecnicos/TecnicosRow.cs
using InPlan.Tenencia;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace InPlan.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("Tecnicos")]
[DisplayName("Técnicos"), InstanceName("Técnico")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[LookupScript]
public sealed partial class TecnicosRow : Row<TecnicosRow.RowFields>, IIdRow, INameRow
{
    const string jIdEmpresa = nameof(jIdEmpresa);
    const string jIdArea = nameof(jIdArea);

    class RowTemplate
    {
        [DisplayName("Id Tecnico"), Identity, IdProperty]
        public int? IdTecnico { get; set; }

        [LookupEditor(typeof(EmpresasRow))]
        [DisplayName("Empresa"), NotNull, ForeignKey("Empresas", "IdEmpresa"), LeftJoin(jIdEmpresa), TextualField(nameof(IdEmpresaDescripcionCorta))]
        public int? IdEmpresa { get; set; }

        [LookupEditor(typeof(AreasRow), CascadeFrom = "IdEmpresa")]
        [DisplayName("Área"), NotNull, ForeignKey("Areas", "IdArea"), LeftJoin(jIdArea), TextualField(nameof(IdAreaDescripcion))]
        public int? IdArea { get; set; }

        [DisplayName("Nombre Completo"), Size(250), NotNull, QuickSearch, NameProperty]
        public string NombreCompleto { get; set; }

        [DisplayName("Celular"), Size(200)]
        public string Celular { get; set; }

        [DisplayName("Email"), Size(255)]
        public string Email { get; set; }

        [DisplayName("Observaciones")]
        public string Observaciones { get; set; }

        [DisplayName("Baja"), NotNull]
        public bool? Baja { get; set; }

        [DisplayName("Empresa"), Expression($"{jIdEmpresa}.[DescripcionCorta]")]
        public string IdEmpresaDescripcionCorta { get; set; }

        [DisplayName("Área"), Expression($"{jIdArea}.[Descripcion]")]
        public string IdAreaDescripcion { get; set; }
    }
}